clusterloader2/testing/list/config.yaml

```diff
@@ -0,0 +1,189 @@
+# List benchmark
+{{$namespaceNumber := DefaultParam .CL2_LIST_NAMESPACE_NUMBER 4}}
+{{$configMapBytes := DefaultParam .CL2_LIST_CONFIG_MAP_BYTES 10000}}
+{{$configMapGroup := DefaultParam .CL2_LIST_CONFIG_MAP_GROUP "list-configmap"}}
+{{$configMapNumber := DefaultParam .CL2_LIST_CONFIG_MAP_NUMBER 3000}}
+{{$secretBytes := DefaultParam .CL2_LIST_SECRET_BYTES 10000}}
+{{$secretGroup := DefaultParam .CL2_LIST_SECRET_GROUP "list-secret"}}
+{{$secretNumber := DefaultParam .CL2_LIST_SECRET_NUMBER 3000}}
+
+{{$listReplicas := DefaultParam .CL2_LIST_BENCHMARK_PODS 1}}
+{{$inflight := DefaultParam .CL2_LIST_CONCURRENCY 2}}
+{{$qps := DefaultParam .CL2_LIST_QPS -1}}
+
+{{$namespacedPodCpu := DefaultParam .CL2_LIST_BENCHMARK_POD_CPU "400m"}}
+{{$namespacedPodMemory := DefaultParam .CL2_LIST_BENCHMARK_POD_MEMORY "800Mi"}}
+{{$clusterScopedPodCpu := DefaultParam .CL2_LIST_BENCHMARK_POD_CPU "2000m"}}
+{{$clusterScopedPodMemory := DefaultParam .CL2_LIST_BENCHMARK_POD_MEMORY "4000Mi"}}
+
+name: List benchmark
+namespace:
+  number: {{$namespaceNumber}}
+tuningSets:
+- name: Sequence
+  parallelismLimitedLoad:
+    parallelismLimit: 10
+steps:
+- name: Setup namespace for list benchmark pods
+  phases:
+  - replicasPerNamespace: 1
+    tuningSet: Sequence
+    objectBundle:
+    - basename: list-benchmark
+      objectTemplatePath: namespace.yaml
```

Comment:
Trying to understand this - so we're creating the objects itself in $namespaceNumber many namespaces and creating a separate namespace to run the pods that list those objects?

Reply:
Correct.
Suppose there are N namespace for creating objects (e.g. configmaps), we will only be able to have N, 2N, 3N ... list benchmarking pods depending on how many replicas of list benchmarking pods per namespace.
But if we separate these 2 kinds of namespaces, we can have any number of list benchmarking pods. IMO this approach is more flexible.